src/Ormolu/Processing/Preprocess.hs

```diff
@@ -126,12 +126,12 @@ magicDisabledLines input =
   where
     go _ [] = []
     go state ((line, i) : ls)
-      | isMagicComment ormoluDisable line,
+      | Just rest <- isMagicComment ormoluDisable line,
         state == OrmoluEnabled =
-          ([i], [(i, magicComment ormoluDisable)]) : go OrmoluDisabled ls
-      | isMagicComment ormoluEnable line,
+          ([i], [(i, magicComment ormoluDisable <> rest)]) : go OrmoluDisabled ls
+      | Just rest <- isMagicComment ormoluEnable line,
         state == OrmoluDisabled =
-          ([i], [(i, magicComment ormoluEnable)]) : go OrmoluEnabled ls
+          ([i], [(i, magicComment ormoluEnable <> rest)]) : go OrmoluEnabled ls
```

Comment:
These two cases contain duplicated logic, so maybe we want to unify them by using sth like 
```haskell
parseMagicComment :: String -> Maybe (OrmoluState, String)
```
instead of the current `isMagicComment`, but I don't think it is crucial to do this just yet.